src/Standards/Generic/Sniffs/ControlStructures/ControlStructuresBracketsNewLineSniff.php

```diff
@@ -0,0 +1,228 @@
+            T_FINALLY,
+        ];
+
+    }//end register()
+
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param integer                     $stackPtr  The position of the current token in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens      = $phpcsFile->getTokens();
+        $errorData   = [strtolower($tokens[$stackPtr]['content'])];
+        $openBrace   = $tokens[$stackPtr]['scope_opener'];
+        $lastContent = $phpcsFile->findPrevious(T_WHITESPACE, ($openBrace - 1), $stackPtr, true);
+        $controlStructureLine = $tokens[$lastContent]['line'];
+        $braceLine            = $tokens[$openBrace]['line'];
+
+        if ($braceLine === $controlStructureLine) {
+            $phpcsFile->recordMetric($stackPtr, 'Control Structure opening brace placement', 'same line');
+            $error = 'Opening brace of a %s must be on the line after the definition';
+            $fix   = $phpcsFile->addFixableError($error, $openBrace, 'OpenBraceNewLine', $errorData);
+
+            if ($fix === true) {
+                $phpcsFile->fixer->beginChangeset();
+
+                if ($tokens[($openBrace - 1)]['code'] === T_WHITESPACE) {
+                    $phpcsFile->fixer->replaceToken(($openBrace - 1), '');
+                }
+
+                $phpcsFile->fixer->addNewlineBefore($openBrace);
+                $phpcsFile->fixer->endChangeset();
+            }
+
+            return;
+        } else {
+            $phpcsFile->recordMetric($stackPtr, 'Control Structure opening brace placement', 'new line');
+
+            if ($braceLine > ($controlStructureLine + 1)) {
+                $error = 'Opening brace of a %s must be on the line following the %s declaration.; Found %s line(s).';
+                $data  = [
+                    $tokens[$stackPtr]['content'],
+                    $tokens[$stackPtr]['content'],
+                    ($braceLine - $controlStructureLine - 1),
+                ];
+                $fix   = $phpcsFile->addFixableError($error, $openBrace, 'OpenBraceWrongLine', $data);
+
+                if ($fix === true) {
+                          $phpcsFile->fixer->beginChangeset();
+
+                    for ($i = ($openBrace - 1); $i > $lastContent; $i--) {
+                        if ($tokens[$i]['line'] === ($tokens[$openBrace]['line'] + 1)) {
+                            break;
+                        }
+
+                        $phpcsFile->fixer->replaceToken($i, '');
+                    }
+
+                          $phpcsFile->fixer->endChangeset();
+                }
+
+                return;
+            }//end if
+        }//end if
+
+        if ($tokens[($openBrace + 1)]['content'] !== $phpcsFile->eolChar) {
+            $error = 'Opening %s brace must be on a line by itself.';
```

Comment:
Sniff errors don't end with full stops
```suggestion
            $error = 'Opening %s brace must be on a line by itself';
```